src/core/pointcloud/qgspointcloudrenderer.cpp

```diff
@@ -159,6 +159,43 @@ QStringList QgsPointCloudRenderer::legendRuleKeys() const
   return QStringList();
 }
 
+void QgsPointCloudRenderer::drawPointToElevationMap( double x, double y, double z, QgsPointCloudRenderContext &context ) const
+{
+  const QPointF originalXY( x, y );
+  context.renderContext().mapToPixel().transformInPlace( x, y );
+  QPainter *elevationPainter = context.elevationPainter();
+  const double zMin = -5000;
+  const double zMax = +5000;
+  float zFloat = ( z - zMin ) / ( zMax - zMin );
+  zFloat = std::clamp<float>( zFloat, 0.0, 1.0 );
+  context.updateZRange( zFloat );
```

Comment:
I'm confused... here the depth values are transformed from [-5000, +5000] to [0,1] but then in applyEyeDomeLighting() they are interpreted as being in [zMin,zMax] range as stored in the context, based on actual values?